File: src/FitnessDietTracker.API/Models/Notification.cs
using FitnessDietTracker.API.Enums;

namespace FitnessDietTracker.API.Models;

public class Notification
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    public NotificationType Type { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public int? RelatedId { get; set; }
    public string? RelatedType { get; set; }
    public NotificationStatus Status { get; set; } = NotificationStatus.Unread;
    public DateTime? ReadAt { get; set; }
    public int? CreatedBy { get; set; }
    public User? Creator { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: server/models/Notification.js
import mongoose from "mongoose";

const notificationSchema = new mongoose.Schema(
  {
    type: {
      type: String,
      enum: [
        "material-missing",
        "material-urgent",
        "finance-change",
        "transfer-status",
        "system-alert",
        "task-assign",
        "other",
      ],
      required: true,
      index: true,
    },
    title: {
      type: String,
      required: true,
    },
    content: {
      type: String,
      required: true,
    },
    priority: {
      type: String,
      enum: ["low", "medium", "high", "urgent"],
      default: "medium",
    },
    recipientIds: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User",
        index: true,
      },
    ],
    readBy: [
      {
        userId: {
          type: mongoose.Schema.Types.ObjectId,
          ref: "User",
        },
        readAt: {
          type: Date,
          default: Date.now,
        },
      },
    ],
    relatedType: String,
    relatedId: {
      type: mongoose.Schema.Types.ObjectId,
      refPath: "relatedType",
    },
    status: {
      type: String,
      enum: ["active", "resolved", "dismissed"],
      default: "active",
      index: true,
    },
    resolvedAt: Date,
    resolvedBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
    },
    resolvedRemark: String,
    pushChannels: [
      {
        type: String,
        enum: ["in-app", "sms", "email", "wechat"],
      },
    ],
    pushedChannels: [
      {
        channel: String,
        pushedAt: Date,
        result: String,
      },
    ],
    createdBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
    },
  },
  {
    timestamps: true,
  }
);

notificationSchema.index({ recipientIds: 1, status: 1, createdAt: -1 });
notificationSchema.index({ type: 1, createdAt: -1 });

export default mongoose.model("Notification", notificationSchema);
